test/sagemaker/test_elastic_inference.py

```diff
@@ -0,0 +1,114 @@
+logging.getLogger('boto3').setLevel(logging.INFO)
+logging.getLogger('botocore').setLevel(logging.INFO)
+logging.getLogger('factory.py').setLevel(logging.INFO)
+logging.getLogger('auth.py').setLevel(logging.INFO)
+logging.getLogger('connectionpool.py').setLevel(logging.INFO)
+logging.getLogger('session.py').setLevel(logging.DEBUG)
+logging.getLogger('functional').setLevel(logging.DEBUG)
+
+
+@pytest.fixture(autouse=True)
+def skip_if_no_accelerator(accelerator_type):
+    if accelerator_type is None:
+        pytest.skip('Skipping because accelerator type was not provided')
+
+
+@pytest.fixture(autouse=True)
+def skip_if_non_supported_ei_region(region):
+    if region not in EI_SUPPORTED_REGIONS:
+        pytest.skip('EI is not supported in {}'.format(region))
+
+
+@pytest.fixture
+def pretrained_model_data(region):
+    return 's3://sagemaker-sample-data-{}/tensorflow/model/resnet/resnet_50_v2_fp32_NCHW.tar.gz'.format(region)
+
+
+def _timestamp():
+    return time.strftime("%Y-%m-%d-%H-%M-%S")
+
+
+def _execution_role(session):
+    return session.resource('iam').Role('SageMakerRole').arn
+
+
+def _production_variants(model_name, instance_type, accelerator_type):
+    production_variants = [{
+        'VariantName': 'AllTraffic',
+        'ModelName': model_name,
+        'InitialInstanceCount': 1,
+        'InstanceType': instance_type,
+        'AcceleratorType': accelerator_type
+    }]
+    return production_variants
+
+
+@pytest.mark.skip_if_non_supported_ei_region
+@pytest.mark.skip_if_no_accelerator
+def test_deploy_elastic_inference_with_pretrained_model(pretrained_model_data,
+                                                        docker_image_uri,
+                                                        instance_type,
+                                                        accelerator_type):
+    endpoint_name = 'test-tfs-ei-deploy-model-{}'.format(_timestamp())
+    endpoint_config_name = 'test-tfs-endpoint-config-{}'.format(_timestamp())
+    model_name = 'test-tfs-ei-model-{}'.format(_timestamp())
+
+    session = boto3.Session()
+    client = session.client('sagemaker')
+    runtime_client = session.client('runtime.sagemaker')
+    client.create_model(ModelName=model_name,
+                        ExecutionRoleArn=_execution_role(session),
+                        PrimaryContainer={
+                            'Image': docker_image_uri,
+                            'ModelDataUrl': pretrained_model_data
+                        })
+
+    logger.info('deploying model to endpoint: {}'.format(endpoint_name))
+
+    client.create_endpoint_config(EndpointConfigName=endpoint_config_name,
+                                  ProductionVariants=_production_variants(model_name, instance_type, accelerator_type))
+
+    client.create_endpoint(EndpointName=endpoint_name,
+                           EndpointConfigName=endpoint_config_name)
+
+    try:
+        client.get_waiter('endpoint_in_service').wait(EndpointName=endpoint_name)
```

Comment:
You don't have to do this, however I can foresee us reusing this logic, maybe it would be better to refactor to another file?